app/views/admin/enterprises/form/_connected_apps.html.haml

```diff
@@ -6,16 +6,14 @@
       %p= t ".tagline"
     %div
       - if enterprise.connected_apps.empty?
-        %button{ data: {reflex: "click->Admin::ConnectedApp#create", enterprise_id: enterprise.id} }
-          = t ".enable"
+        = button_to t(".enable"), admin_enterprise_connected_apps_path(enterprise.id), method: :post
```

Comment:
> Technically, the new buttons are a form within a form which is invalid HTML, but it works.

That doesn't seem ideal to me, and might have unexpected drawbacks. But I can't think of a better, so if it works, it works!

I'm guessing if you have other unsaved changes on the Enterprise (from another tab), it will display a confirmation as usual? That could be a little confusing for users, but I think that would be rare and fine for this case.
I'm not sure if the enterprise fields would also be submitted by this button, but even if they are, I expect they'd be silently ignored.

Reply:
This may be the cause for the latest bug:

* #12533

My tests show that it was this commit.